homeassistant/components/trafikverket_camera/__init__.py

```diff
@@ -30,3 +38,37 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    # Change entry unique id from location to camera id
+    if entry.version == 1:
+        location = entry.data[CONF_LOCATION]
+        api_key = entry.data[CONF_API_KEY]
+
+        web_session = async_get_clientsession(hass)
+        camera_api = TrafikverketCamera(web_session, api_key)
+
+        try:
+            camera_info = await camera_api.async_get_camera(location)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error(
+                "Could not migrate the config entry. No connection to the api"
+            )
+            return False
+
+        if camera_id := camera_info.camera_id:
+            entry.version = 2
+            _LOGGER.debug(
+                "Migrate Trafikverket Camera config entry unique id to %s",
+                camera_id,
+            )
+            hass.config_entries.async_update_entry(
+                entry,
+                unique_id=f"{DOMAIN}-{camera_id}",
+            )
+            return True
+        _LOGGER.error("Could not migrate the config entry. Camera has no id")
```

Comment:
How many cameras have no id?
If a user uses a camera without an id -> the migration will fail always. Therefore the integration with those cameras is not longer usable, which is a breaking change.

Reply:
They always have an id. This is more precaution in case the `CameraInfo` is loaded incorrectly e.g. missing id.
Should never happen.